docker/qgisserver/geomapfish_qgisserver/accesscontrol.py

```diff
@@ -182,42 +186,47 @@ class OGCServerAccessControl(QgsAccessControlFilter):
 
     SUBSETSTRING_TYPE = ["PostgreSQL database with PostGIS extension"]
 
-    def __init__(self, server_iface, ogcserver_name, srid, DBSession):  # noqa: N803
+    def __init__(self, server_iface, ogcserver_name, map_file, srid, DBSession):  # noqa: N803
         super().__init__(server_iface)
 
         self.server_iface = server_iface
         self.DBSession = DBSession
         self.area_cache = {}
         self.layers = None
         self.lock = Lock()
-        self.project = None
         self.srid = srid
         self.ogcserver = None
 
-        try:
-            from c2cgeoportal_commons.models import InvalidateCacheEvent
+        from c2cgeoportal_commons.models import InvalidateCacheEvent
 
-            @zope.event.classhandler.handler(InvalidateCacheEvent)
-            def handle(event: InvalidateCacheEvent):  # pylint: disable=unused-variable
-                del event
-                LOG.info("=== invalidate ===")
-                with self.lock:
-                    self.layers = None
+        @zope.event.classhandler.handler(InvalidateCacheEvent)
+        def handle(_: InvalidateCacheEvent):  # pylint: disable=unused-variable
+            LOG.info("=== invalidate ===")
+            self._init(ogcserver_name, map_file)
 
-            from c2cgeoportal_commons.models.main import OGCServer
+        config_cache = QgsConfigCache.instance()
+        self.project = config_cache.project(map_file)
+        self._init(ogcserver_name, map_file)
 
-            self.ogcserver = (
-                self.DBSession.query(OGCServer).filter(OGCServer.name == ogcserver_name).one_or_none()
-            )
-            if self.ogcserver is None:
-                LOG.warning("No OGC server found for '{}' => no rights".format(ogcserver_name))
+    def _init(self, ogcserver_name, map_file):
+        with self.lock:
+            try:
+                config_cache = QgsConfigCache.instance()
+                self.project = config_cache.project(map_file)
```

Comment:
This seems already done some lines above

Reply:
You comment an old commit...

Reply:
Maybe I miss something, but I see the same lines at 208 and 209 and the page is refreshed. Maybe an error from github.

Reply:
Now it's working...
Effectively I do that to remove a warning in the IDE, I need to initialize project in the `__init__` method and if I set it to None I get an error in the IDE because he considers that he can be None.
If you think that's durty I can remove it...